File: apps/api/src/lib/minio.ts
import * as Minio from 'minio';
import { ENV } from '../config/env';

export const minio = new Minio.Client({
  endPoint: ENV.MINIO_ENDPOINT,
  port: ENV.MINIO_PORT,
  useSSL: false,
  accessKey: ENV.MINIO_ACCESS_KEY,
  secretKey: ENV.MINIO_SECRET_KEY,
});

export async function ensureMinioBucket(): Promise<void> {
  try {
    const exists = await minio.bucketExists(ENV.MINIO_BUCKET);
    if (!exists) {
      await minio.makeBucket(ENV.MINIO_BUCKET, 'us-east-1');
      console.log(`[MinIO] bucket '${ENV.MINIO_BUCKET}' created`);
    }
  } catch (err) {
    console.warn('[MinIO] could not ensure bucket (skipping):', (err as Error).message);
  }
}

File: apps/api/internal/core/minio/minio.go
// Package minio provides a MinIO client factory.
package minio

import (
	"context"
	"fmt"
	"log"

	"github.com/minio/minio-go/v7"
	"github.com/minio/minio-go/v7/pkg/credentials"
)

// Connect creates a MinIO client.
func Connect(endpoint string, port int, accessKey, secretKey string, useSSL bool) (*minio.Client, error) {
	return minio.New(fmt.Sprintf("%s:%d", endpoint, port), &minio.Options{
		Creds:  credentials.NewStaticV4(accessKey, secretKey, ""),
		Secure: useSSL,
	})
}

// EnsureBucket creates bucket if it doesn't exist.
func EnsureBucket(ctx context.Context, client *minio.Client, bucket string) {
	exists, err := client.BucketExists(ctx, bucket)
	if err != nil || exists {
		return
	}
	if err := client.MakeBucket(ctx, bucket, minio.MakeBucketOptions{Region: "us-east-1"}); err != nil {
		log.Printf("minio: failed to create bucket %s: %v", bucket, err)
	}
}
